### ISSUE TITLE
softmax vs cosine softmax ? 

### ISSUE THREAD

**Issue by youngjae-avikus** [ZQPei/deep_sort_pytorch#257]
Hi 
In the official code (https://github.com/nwojke/deep_sort), he focused on generative learning using cosine_softmax loss
![image](https://user-images.githubusercontent.com/99321359/206056647-36147851-fd6b-4eb9-a090-27cdbdd356d3.png)

However, in the code implemented by pytorch that code refer to, it simply conducted a discrimitive learning with cross_entropy

https://github.com/ZQPei/deep_sort_pytorch/blob/8cfe2467a4b1f4421ebf9fcbc157921144ffe7cf/deep_sort/deep/train.py#L67

Is there a big difference between the reason and the performance of this implementation?

**Comment by yesmineht** [comment]
hello @youngjae-avikus did you change it to cosine softmax_loss ?
